### PR TITLE
Move server-io

### PR DESCRIPTION
**Pull request by AlexTugarev**

### CODE REVIEW

File: chart/templates/proxy-configmap.yaml [hunk 1]
```diff
@@ -48,3 +48,51 @@ data:
...
+        import upstream_headers
+        import upstream_connection
+
+        flush_interval -1
+      }
+
+    }
+{{- end }}
+{{- if not .Values.components.paymentEndpoint.disabled }}
+  vhost.kedge: |
```
Comment (geropl) [inline chart/templates/proxy-configmap.yaml:74]: `vhost.paypent-endpoint` or similar

File: chart/values.yaml [hunk 2]
```diff
@@ -303,6 +303,9 @@ components:
       disabled: "false"
     definitelyGpDisabled: "false"
 
+  paymentEndpoint:
```
Comment (geropl) [inline chart/values.yaml:306]: `kedge` is not necessary either, right?

File: components/dashboard/src/settings/Plans.tsx [hunk 3]
```diff
@@ -131,7 +131,7 @@ export default function () {
         } else if (!pollAccountStatementTimeout) {
             // Refresh account statement in 5 seconds in order to poll for upgrade confirmed
             pollAccountStatementTimeout = setTimeout(async () => {
-                const statement = await server.getAccountStatement({});
+                const statement = await getGitpodService().server.getAccountStatement({});
```
Comment (geropl) [inline components/dashboard/src/settings/Plans.tsx:134]: Why not re-use `server` here?

File: components/ee/payment-endpoint/test/fixtures/downgrade_unlimited_to_professional_new/01.downgrade-change-actual.json [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+      "due_invoices_count": 0,
+      "mrr": 2917,
+      "exchange_rate": 1.0,
+      "base_currency_code": "EUR"
+    },
+    "customer": {
+      "id": "643ac637-74ae-4f82-9a86-c8527eb1e496",
+      "first_name": "Jan",
+      "last_name": "Keromnes",
+      "email": "jan.keromnes@typefox.io",
```
Comment (geropl) [inline components/ee/payment-endpoint/test/fixtures/downgrade_unlimited_to_professional_new/01.downgrade-change-actual.json:37]: IMO the name and email are the only things we need to purge here. Everything else is random test data.

File: components/ee/payment-endpoint/test/fixtures/downgrade_unlimited_to_professional_new/01.downgrade-change-actual.json [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+      "created_from_ip": "37.173.245.142",
+      "taxability": "taxable",
+      "updated_at": 1559560635,
+      "pii_cleared": "active",
+      "resource_version": 1559560635963,
+      "deleted": false,
+      "object": "customer",
+      "billing_address": {
+        "first_name": "Jan",
+        "last_name": "Keromnes",
```
Comment (geropl) [inline components/ee/payment-endpoint/test/fixtures/downgrade_unlimited_to_professional_new/01.downgrade-change-actual.json:52]: Here as well.

File: components/ee/payment-endpoint/test/fixtures/first_basic/01.basic-created.json [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+            "resource_version": 1543492872468,
+            "deleted": false,
+            "object": "subscription",
+            "currency_code": "USD",
+            "due_invoices_count": 0,
+            "mrr": 0
+        },
+        "customer": {
+            "id": "68b75ca3-df45-4fde-8665-9f9f3d749d55",
+            "email": "gero.posmyk-leinemann@typefox.io",
```
Comment (geropl) [inline components/ee/payment-endpoint/test/fixtures/first_basic/01.basic-created.json:37]: same

File: components/ee/payment-endpoint/test/fixtures/first_basic/01.basic-created.json [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+            "pii_cleared": "active",
+            "resource_version": 1543492870889,
+            "deleted": false,
+            "object": "customer",
+            "billing_address": {
+                "first_name": "test",
+                "last_name": "test2",
+                "line1": "Uelzener Straße  1",
+                "city": "Lüneburg",
+                "state": "Niedersachsen",
```
Comment (geropl) [inline components/ee/payment-endpoint/test/fixtures/first_basic/01.basic-created.json:55]: Test data, but we should remove nonetheless.

File: components/ee/payment-endpoint/test/fixtures/first_basic/01.basic-created.json [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+            "unbilled_charges": 0,
+            "preferred_currency_code": "USD",
+            "cs_typefox___privacy": true
+        },
+        "card": {
+            "status": "valid",
+            "gateway": "stripe",
+            "gateway_account_id": "gw_B3ikeGR3Q7TomGst",
+            "iin": "******",
+            "last4": "4242",
```
Comment (geropl) [inline components/ee/payment-endpoint/test/fixtures/first_basic/01.basic-created.json:83]: to clarify: Credit cards are none-existent test data, too.

File: components/gitpod-db/src/typeorm/migration/1621920691254-InitialComSetup.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+ * See License.enterprise.txt in the project root folder.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class InitialComSetup1595232295133 implements MigrationInterface {
+
+    public async up(queryRunner: QueryRunner): Promise<any> {
+        // TODO: FIXME remove these test queries
+        await queryRunner.query("CREATE TABLE IF NOT EXISTS `d_b_account_entry` (`id` int(11) DEFAULT NULL, `userId` char(36) NOT NULL, `amount` double NOT NULL, `date` varchar(255) NOT NULL, `expiryDate` varchar(255) NOT NULL DEFAULT '', `kind` char(7) NOT NULL, `description` text, `uid` char(36) NOT NULL, `creditId` char(36) DEFAULT NULL, `_lastModified` timestamp(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6), PRIMARY KEY (`uid`), KEY `ind_dbsync` (`_lastModified`), KEY `ind_expiryDate` (`expiryDate`)) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;");
```
Comment (geropl) [inline components/gitpod-db/src/typeorm/migration/1621920691254-InitialComSetup.ts:13]: Is this taken from the live prod/staging DB? Or from source?

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 10]
```diff
@@ -314,6 +322,9 @@ export class GitpodServerImpl<Client extends GitpodClient, Server extends Gitpod
                 result.push(info);
                 continue;
             }
+            if (builtinAuthProvidersConfigured && !isBuiltIn(info)) {
```
Comment (geropl) [inline components/server/src/workspace/gitpod-server-impl.ts:325]: :+1:

**Comment by geropl** [comment]
@AlexTugarev I make a quick first round. Looks good so far sans the minor comments.

If I should check sth specifically please ping again. :upside_down_face:

File: components/gitpod-db/src/typeorm/migration/1621920691254-InitialComSetup.ts (hunk 9 above)
Comment (AlexTugarev) [reply components/gitpod-db/src/typeorm/migration/1621920691254-InitialComSetup.ts:13]: from source. do you expect differences?

File: components/dashboard/src/settings/Plans.tsx (hunk 3 above)
Comment (AlexTugarev) [reply components/dashboard/src/settings/Plans.tsx:134]: `server` ist the `json-rpc` proxy interface, which on reconnects of websocket might get obsolete.

File: chart/templates/proxy-configmap.yaml (hunk 1 above)
Comment (AlexTugarev) [reply chart/templates/proxy-configmap.yaml:74]: nice catch!

File: components/proxy/conf/Caddyfile [hunk 11]
```diff
@@ -186,6 +186,27 @@ https://{$GITPOD_DOMAIN} {
 	}
 }
 
+https://payment.{$GITPOD_DOMAIN} {
```
Comment (aledbf) [inline components/proxy/conf/Caddyfile:189]: @AlexTugarev please move the block to https://github.com/gitpod-io/gitpod/blob/main/chart/templates/proxy-configmap.yaml#L16 (not sure payment is always enabled or not)

Comment (csweichel) [reply components/proxy/conf/Caddyfile]: fixed

**Review by csweichel — APPROVED** [review]
LGTM

**Pull request merged**